connectors/sources/jira.py

```diff
@@ -390,19 +430,29 @@ async def _get_projects(self):
         try:
             timezone = await self._get_timezone()
 
-            async for response in self.jira_client.api_call(url_name=PROJECT):
-                response = await response.json()
-                for project in response:
-                    yield {
-                        "_id": f"{project['name']}-{project['id']}",
-                        "_timestamp": iso_utc(
-                            when=datetime.now(
-                                pytz.timezone(timezone)  # pyright: ignore
-                            )
-                        ),
-                        "Type": "Project",
-                        "Project": project,
-                    }
+            _timestamp = iso_utc(when=datetime.now(pytz.timezone(timezone)))
+            if self.jira_client.projects == ["*"]:
+                async for response in self.jira_client.api_call(url_name=PROJECT):
+                    response = await response.json()
+                    for project in response:
+                        yield {
+                            "_id": f"{project['name']}-{project['id']}",
+                            "_timestamp": _timestamp,
+                            "Type": "Project",
+                            "Project": project,
+                        }
+            else:
+                for project_key in self.jira_client.projects:
+                    async for response in self.jira_client.api_call(
+                        url_name=PROJECT_BY_KEY, key=project_key
+                    ):
+                        project = await response.json()
+                        yield {
+                            "_id": f"{project['name']}-{project['id']}",
+                            "_timestamp": _timestamp,
+                            "Type": "Project",
+                            "Project": project,
```

Comment:
We've already discussed regarding this a call and mentioned in TDD as well. Since, Jira supports custom fields, so we would never know to which field to index and not. That's why we came on a point to add everything getting from an API response.
cc: @danajuratoni

Reply:
@danajuratoni and I are following up on this offline. We can come back after the fact to make changes if necessary, but I won't block this PR on that ongoing discussion.